src/main/java/uk/gov/ons/ctp/response/sample/scheduled/distribution/SampleUnitDistributor.java

```diff
@@ -1,170 +1,122 @@
-                e.getMessage(),
-                sampleUnit.getSampleUnitPK());
-            log.error("Stack trace: " + e);
-            failures++;
-          }
-        }
-        sampleUnitDistributionListManager.deleteList(SAMPLEUNIT_DISTRIBUTOR_LIST_ID, true);
+      for (uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit msu : mappedSampleUnits) {
         try {
-          sampleUnitDistributionListManager.unlockContainer();
-        } catch (LockingException le) {
-          // oh well - will time out or we never had the lock
+          sampleUnitSender.sendSampleUnit(msu);
+        } catch (CTPException e) {
+          hasErrors = true;
+          log.error(
+              "Failed to send a sample unit to queue and update state with ID: {}", msu.getId(), e);
         }
       }
-    } catch (Exception e) {
-      log.error("Failed to process sample units", e);
+
+      if (!hasErrors) {
+        job.setJobComplete(true);
+        collectionExerciseJobRepository.saveAndFlush(job);
+      }
+    } finally {
+      // Finally, we can unlock the distributed lock
+      lock.unlock();
     }
+  }
 
-    distInfo.setSampleUnitsSucceeded(successes);
-    distInfo.setSampleUnitsFailed(failures);
+  private List<uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit> getMappedSampleUnits(
+      UUID sampleSummaryId, String collectionExerciseId) {
 
-    log.info("SampleUnitsDistributor sleeping");
-    return distInfo;
-  }
+    List<uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit> mappedSampleUnits =
+        new LinkedList<>();
 
-  private SampleAttributes mapSampleAttributes(
-      uk.gov.ons.ctp.response.sample.domain.model.SampleAttributes sampleAttributes) {
-    SampleAttributes mappedSampleAttributes = new SampleAttributes();
-    SampleAttributes.Builder<Void> sampleAttributesBuilder =
-        mappedSampleAttributes.newCopyBuilder();
-    for (Map.Entry<String, String> attribute : sampleAttributes.getAttributes().entrySet()) {
-      sampleAttributesBuilder
-          .addEntries()
-          .withKey(attribute.getKey())
-          .withValue(attribute.getValue());
-    }
-    return sampleAttributesBuilder.build();
-  }
+    SampleSummary sampleSummary = sampleSummaryRepository.findById(sampleSummaryId);
 
-  /**
-   * Sends SampleUnit to the CollectionExercise queue
-   *
-   * @param sampleUnit sample unit to be mapped
-   * @param mappedSampleUnit mapped sample unit to be sent
-   * @throws CTPException if transition issue
-   */
-  private void sendSampleUnitToCollectionExerciseQueue(
-      SampleUnit sampleUnit,
-      uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit mappedSampleUnit)
-      throws CTPException {
-    transitionSampleUnitStateFromDeliveryEvent(sampleUnit);
-    sampleUnitPublisher.send(mappedSampleUnit);
-  }
+    if (sampleSummary.getState() == SampleState.ACTIVE) {
+      try (Stream<SampleUnit> sampleUnits =
```

Comment:
Looks like SampleUnit only gets used once where as uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit gets used multiple times is it worth switching the imports around?